tests/Doctrine/Tests/ORM/Functional/AdvancedDqlQueryWithCollationTest.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional;
+
+use Doctrine\DBAL\Platforms\MySQLPlatform;
+use Doctrine\Tests\Models\ProductsRomanCollation\ProductColor;
+use Doctrine\Tests\OrmFunctionalTestCase;
+use Doctrine\Tests\TestUtil;
+
+/**
+ * Functional Query tests.
+ */
+class AdvancedDqlQueryWithCollationTest extends OrmFunctionalTestCase
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @see \Doctrine\Tests\OrmFunctionalTestCase::$sharedConn
+     */
+    protected static $sharedConn;
+
+    /**
+     * {@inheritdoc}
+     *
+     * @see \PHPUnit\Framework\TestCase::$backupGlobals
+     */
+    protected $backupGlobals = true;
+
+    /**
+     * {@inheritdoc}
+     *
+     * @see \Doctrine\Tests\OrmFunctionalTestCase::setUp()
+     */
+    protected function setUp(): void
+    {
+        static::$sharedConn = TestUtil::getConnection();
+        if (! static::$sharedConn->getDatabasePlatform() instanceof MySQLPlatform) {
+            self::markTestSkipped('The ' . self::class . ' requires the use of mysql.');
+        }
+
+        static::$sharedConn    = null;
+        $GLOBALS['db_charset'] = 'utf8';
```

Comment:
> This will affect all tests after this with a new connection

Nope, `$sharedConn` is specific to AdvancedDqlQueryWithCollationTest, and other tests use another $sharedConn (so their connection is different from the one used in `AdvancedDqlQueryWithCollationTest`.

Reply:
> Couldn't charset and collation be set in the table options of the entity instead?

I wasn't aware that `@Table` annotations support `options` (there's no mention to that [in the docs](https://www.doctrine-project.org/projects/doctrine-orm/en/2.11/reference/annotations-reference.html#table).
But yep, by checking [the PHP code](https://github.com/doctrine/orm/blob/93f9eb7af287762c991c553d23f84053c092f90f/lib/Doctrine/ORM/Mapping/Table.php#L31) it seems it's possible.

Reply:
I've rewritten the new `AdvancedDqlQueryWithCollationTest` test by using the (much simpler) `@Table(..., options={...})` approach.